symmetric-core/src/main/java/org/jumpmind/symmetric/util/PropertiesUtil.java

```diff
@@ -41,14 +43,27 @@ public static String getEnginesDir() {
         return enginesDir;
     }
 
-    public static File findPropertiesFileForEngineWithName(String engineName) {
+    public static File findPropertiesFileForEngineWithName(String engineName, Map<String, String> replacementValues) {
+        if (replacementValues == null) {
+            replacementValues = new HashMap<String, String>();
+        }
+        boolean replacementValuesProvided = !replacementValues.isEmpty();
         File[] files = findEnginePropertiesFiles();
```

Comment:
Avoid the double negative tests and just call this variable something like isEmptyReplacementValues = replacementValues.isEmpty().